src/ethereum_test/types.py

```diff
@@ -265,11 +286,23 @@ def __post_init__(self) -> None:
         if self.signature is None and self.secret_key is None:
             self.secret_key = TestPrivateKey
 
+        if self.ty is None:
+            # Try to deduce transaction type from included fields
+            if (
+                self.max_fee_per_gas is not None
```

Comment:
If only one is set, will the other be set automatically? I think yes?

Reply:
I think it will fail, I checked geth code and it requires both, since there is no way to deduce the other value if only one is set I think.